internal/provider/schema_gitlab_project_issue.go

```diff
@@ -0,0 +1,364 @@
+			Optional:         true,
+			Default:          "issue",
+			ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice(validIssueTypes, false)),
+		},
+		"labels": {
+			Description: "The labels of an issue.",
+			Type:        schema.TypeSet,
+			Elem:        &schema.Schema{Type: schema.TypeString},
+			Optional:    true,
+		},
+		"merge_request_to_resolve_discussions_of": {
+			Description: "The IID of a merge request in which to resolve all issues. This fills out the issue with a default description and mark all discussions as resolved. When passing a description or title, these values take precedence over the default values.",
+			Type:        schema.TypeInt,
+			Optional:    true,
+		},
+		"milestone_id": {
+			Description: "The global ID of a milestone to assign issue. To find the milestone_id associated with a milestone, view an issue with the milestone assigned and use the API to retrieve the issue's details.",
+			Type:        schema.TypeInt,
+			Optional:    true,
+		},
+		"weight": {
+			Description:      "The weight of the issue. Valid values are greater than or equal to 0.",
+			Type:             schema.TypeInt,
+			Optional:         true,
+			Computed:         true,
+			ValidateDiagFunc: validation.ToDiagFunc(validation.IntAtLeast(0)),
+		},
+		"external_id": {
+			Description: "The external ID of the issue.",
+			Type:        schema.TypeString,
+			Computed:    true,
+		},
+		// NOTE: not part of `CREATE`, but part of `UPDATE` with the `state_event` field.
+		"state": {
+			Description:      fmt.Sprintf("The state of the issue. Valid values are: %s.", renderValueListForDocs(validIssueStates)),
+			Type:             schema.TypeString,
+			Optional:         true,
+			Default:          "opened",
+			ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice(validIssueStates, false)),
+		},
+		// NOTE: it keep things simple users of this resource should use the `gitlab_user` data source to
+		//       get more information about the author if desired.
+		"author_id": {
+			Description: "The id of the author of the issue. Use `gitlab_user` data source to get more information about the user.",
+			Type:        schema.TypeInt,
+			Computed:    true,
+		},
+		"updated_at": {
+			Description: "When the issue was last. Date time string, ISO 8601 formatted, for example 2016-03-11T03:45:40Z.",
+			Type:        schema.TypeString,
+			Computed:    true,
+			Optional:    true,
+		},
+		"closed_at": {
+			Description: "When the issue was closed. Date time string, ISO 8601 formatted, for example 2016-03-11T03:45:40Z.",
+			Type:        schema.TypeString,
+			Computed:    true,
+			Optional:    true,
+		},
+		// NOTE: it keep things simple users of this resource should use the `gitlab_user` data source to
+		//       get more information about the closer if desired.
+		"closed_by_user_id": {
+			Description: "The id of the user that closed the issue. Use `gitlab_user` data source to get more information about the user.",
```

Comment:
nit: Noticing inconsistencies between using "id" and "ID" in descriptions. Not sure which is preferred.